tests/unit/test_strategy_registry.py

```
@@ -0,0 +1,332 @@
+def test_json_pairs_rejects_non_array_and_bad_pair_shape() -> None:
+    with pytest.raises(TypeError, match="JSON array of pairs"):
+        _json_pairs("bad")
+    with pytest.raises(TypeError, match="JSON array pair"):
+        _json_pairs([["ok", 1], ["missing-value"]])
+
+
+def test_json_string_and_string_list_validate_types() -> None:
+    assert _json_string("default", "config.strategy_id") == "default"
+    assert _json_string_list(["brier", "pnl"], "eval_spec.metrics") == ["brier", "pnl"]
+
+    with pytest.raises(TypeError, match="config.strategy_id must decode to a string"):
```

Ruff (0.15.10)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)